### PR TITLE
WIP: status-ng: Emojify "start", "stop", and "delete" & centralize container runtime handling

### PR DESCRIPTION
**Pull request by tstromberg**
This is the initial PR for implementing the status-ng design doc: https://docs.google.com/document/d/1UfTPbQx-WLphtyR-G-Zj3Ekcpoe9PjYEV4srgbCru7U/edit

Notably absent are colorized text or a status spinner. Those will come later. 

This PR centralizes the container runtime code into a single package, because I didn't want to duplicate it's atrocities elsewhere. This also adds a PullImages call to the bootstrapper interface, so that the UI can separately trigger and communicate the lengthy download process from the less-lengthy startup process.

### CODE REVIEW

**Comment by tstromberg** [comment]
NOTE: The test failures indicate that there is currently a bug in how containerd is handled. Probably just an inverted boolean check.

**Comment by afbjorklund** [comment]
It would probably be a good idea to concentrate all the emojis in one place, with something like `_()` ?

* i.e. `gettext` : https://www.gnu.org/software/gettext/manual/html_node/Mark-Keywords.html

That one is used for translating messages rather than emojifying them, but anyway. Same approach.

That would make it easy to go back to ASCII, if the current terminal/user isn't up to the challenge...

File: cmd/minikube/cmd/start.go [hunk 1]
```diff
@@ -172,18 +179,33 @@ func runStart(cmd *cobra.Command, args []string) {
...
-
+	clusterConfig := cfg.Config{MachineConfig: config}
 	if err := saveConfig(clusterConfig); err != nil {
 		glog.Errorln("Error saving profile cluster configuration: ", err)
 	}
 
-	fmt.Printf("Starting local Kubernetes %s cluster...\n", viper.GetString(kubernetesVersion))
-	fmt.Println("Starting VM...")
+	if config.VMDriver == "none" {
+		// TODO(tstromberg): Hide this warning if
```
Comment (balopat) [inline cmd/minikube/cmd/start.go]: does this TODO still apply?

File: cmd/minikube/cmd/config/validations.go [hunk 2]
```diff
@@ -128,6 +128,7 @@ func IsValidAddon(name string, val string) error {
 	return errors.Errorf("Cannot enable/disable invalid addon %s", name)
 }
 
+// IsContainerdRuntime is a validator which returns an error if the current runtime is not containerd
 func IsContainerdRuntime(_, _ string) error {
```
Comment (RA489) [inline cmd/minikube/cmd/config/validations.go]: Is it currently support containerd as runtime? Ideally minikube should respect whatever container runtimes may be

**Comment by tstromberg** [comment]
Thanks for the feedback! I'm going to split this PR up into a couple of them for sanity sake.

**Comment by afbjorklund** [comment]
Think the split (emojis/runtime/logging) needs to happen _first_, before this PR can solve separate issues.

**Pull request closed without merging**